### PR TITLE
Port of web_action_conditionable to 10.0

### PR DESCRIPTION
**Pull request by daramousk**

### CODE REVIEW

File: web_action_conditionable/README.rst [hunk 1]
```diff
@@ -40,10 +40,10 @@ To use this module, you need to:
 
 .. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
    :alt: Try me on Runbot
-   :target: https://runbot.odoo-community.org/runbot/web/8.0
+   :target: https://runbot.odoo-community.org/runbot/web/10.0
 
 .. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
```
Comment (pedrobaeza) [inline web_action_conditionable/README.rst]: You can remove this comment

File: web_action_conditionable/README.rst [hunk 2]
```diff
@@ -60,7 +60,7 @@ Bug Tracker
 Bugs are tracked on `GitHub Issues <https://github.com/OCA/web/issues>`_.
 In case of trouble, please check there if your issue has already been reported.
 If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
-`here <https://github.com/OCA/web/issues/new?body=module:web_action_conditionableversion:%208.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
+`here <https://github.com/OCA/web/issues/new?body=module:web_action_conditionableversion:%2010.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (pedrobaeza) [inline web_action_conditionable/README.rst]: Use short version as new template: https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

File: web_action_conditionable/README.rst [hunk 3]
```diff
@@ -40,10 +40,10 @@ To use this module, you need to:
```
Comment (pedrobaeza) [inline web_action_conditionable/README.rst:37]: Remove initial sentence, as it's not filled.

File: web_action_conditionable/views/view.xml [hunk 4]
```diff
@@ -1,10 +1,8 @@
...
-        </template>
-    </data>
-</openerp>
+<odoo>
+	<template id="assets_backend" name="action conditionable assets" inherit_id="web.assets_backend">
+		<xpath expr="." position="inside">
+			<script type="text/javascript" src="/web_action_conditionable/static/src/js/views.js"></script>
+		</xpath>
+	</template>
+</odoo>
```
Comment (pedrobaeza) [inline web_action_conditionable/views/view.xml]: End line break.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by daramousk** [comment]
@pedrobaeza Further commit has been done, I have tried to address some Codacy issues as well as implement the changes requested.

Thanks

File: web_action_conditionable/README.rst [hunk 5]
```diff
@@ -15,7 +15,7 @@ with this module you can:
 
 ::
    
-   <tree delete="state=='draft'">
+   <tree delete="'state'=='draft'">
```
Comment (yajo) [inline web_action_conditionable/README.rst:18]: `'state'` will absolutely never be `'draft'`; leave this as it was 😉

File: web_action_conditionable/static/src/js/views.js [hunk 6]
```diff
@@ -1,22 +1,21 @@
...
+    var view = require("web.View");
+    view.include({
+	is_action_enabled : function(action) {
+	    var attrs = this.fields_view.arch.attrs;
+	    if (action in attrs) {
+		try {
+		    return this._super(action);
+		} catch (error) {
+		    var expr = attrs[action];
+		    var expression = py.parse(py.tokenize(expr));
```
Comment (yajo) [inline web_action_conditionable/static/src/js/views.js]: You need to `require()` what provides `py` at the top of the module.

File: web_action_conditionable/README.rst [hunk 7]
```diff
@@ -34,16 +34,14 @@ Not needed.
...
 
 * go to ...
 
 .. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
    :alt: Try me on Runbot
-   :target: https://runbot.odoo-community.org/runbot/web/8.0
 
 .. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
-.. branch is "8.0" for example
+.. branch is "10.0" for example
```
Comment (yajo) [inline web_action_conditionable/README.rst]: don't fix this comment; better, follow what it tells you to do.

File: web_action_conditionable/README.rst [hunk 8]
```diff
@@ -34,16 +34,14 @@ Not needed.
 Usage
 =====
 
-To use this module, you need to:
 
 * go to ...
 
 .. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
    :alt: Try me on Runbot
-   :target: https://runbot.odoo-community.org/runbot/web/8.0
```
Comment (yajo) [inline web_action_conditionable/README.rst:43]: You must provide a good link, not remove it.

File: web_action_conditionable/static/src/js/views.js [hunk 9]
```diff
@@ -1,22 +1,21 @@
-/*global openerp, _, $ */
-
-openerp.web_action_conditionable = function (instance) {
-  instance.web.View.include({
-    is_action_enabled: function(action) {
-      var attrs = this.fields_view.arch.attrs;
-      if (action in attrs) {
-	try {
-	  return this._super(action);
```
Comment (yajo) [inline web_action_conditionable/static/src/js/views.js:9]: Why did you remove this part? If `this._super` does something, we want to follow that first, no? Or is there no super?

File: web_action_conditionable/views/view.xml [hunk 10]
```diff
@@ -1,10 +1,8 @@
...
-            </xpath>
-        </template>
-    </data>
-</openerp>
+<odoo>
+	<template id="assets_backend" name="action conditionable assets" inherit_id="web.assets_backend">
+		<xpath expr="." position="inside">
+			<script type="text/javascript" src="/web_action_conditionable/static/src/js/views.js"></script>
+		</xpath>
+	</template>
```
Comment (yajo) [inline web_action_conditionable/views/view.xml]: OCA standard is to indent with 4 spaces, please leave it as it was.

File: web_action_conditionable/static/src/js/views.js [hunk 11]
```diff
@@ -1,22 +1,21 @@
...
+	    if (action in attrs) {
+		try {
+		    return this._super(action);
+		} catch (error) {
+		    var expr = attrs[action];
+		    var expression = py.parse(py.tokenize(expr));
+		    var result = py.evaluate(expression).toJSON();
+		    return result;
+		}
+	    } else {
```
Comment (yajo) [inline web_action_conditionable/static/src/js/views.js]: Never merge spaces with tabs. No flamewars here, it's just OCA standard to use 4 spaces, so please follow it 😊

**Review by yajo — CHANGES_REQUESTED** [review]
It would be awesome if you provide some demo data for functional tests, but at least there are several comments that need fixing.

**Review by jesusVMayor — APPROVED** [review]

**Review by jesusVMayor — COMMENTED** [review]
Wrong button

File: web_action_conditionable/static/src/js/views.js (hunk 6 above)
Comment (daramousk) [reply web_action_conditionable/static/src/js/views.js]: py Seems to already be in the global namespace as declared on `web/static/lib/py.js/lib/py/js` and whenever is used in the base code in Odoo, require.js is not used. Correct me if I am wrong here since I do not have frontend experience.

File: web_action_conditionable/README.rst (hunk 5 above)
Comment (daramousk) [reply web_action_conditionable/README.rst:18]: It will not work if I leave it as it was on 10.0

**Comment by daramousk** [comment]
@Yajo Further changes pushed

**Comment by daramousk** [comment]
@Yajo Let me know what you think about this one as well.

File: web_action_conditionable/views/view.xml [hunk 12]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="utf-8"?>
-<odoo>
-	<template id="assets_backend" name="action conditionable assets" inherit_id="web.assets_backend">
-		<xpath expr="." position="inside">
-			<script type="text/javascript" src="/web_action_conditionable/static/src/js/views.js"></script>
-		</xpath>
-	</template>
-</odoo>
+<openerp>
+    <data>
```
Comment (yajo) [inline web_action_conditionable/views/view.xml]: When I said "leave it as it was", I was speaking about indentation, not about these deprecated tags...

File: web_action_conditionable/README.rst (hunk 3 above)
Comment (yajo) [reply web_action_conditionable/README.rst:37]: Instead of removing, you better add usage instructions, because now I don't know what should I test...

**Review by yajo — CHANGES_REQUESTED** [review]

File: web_action_conditionable/static/src/js/views.js [hunk 13]
```diff
@@ -1,22 +1,22 @@
...
-  instance.web.View.include({
-    is_action_enabled: function(action) {
-      var attrs = this.fields_view.arch.attrs;
-      if (action in attrs) {
-	try {
-	  return this._super(action);
-	} catch(error) {
-	  var expr = attrs[action];
-	  var expression = py.parse(py.tokenize(expr));
-	  var cxt = this.dataset.get_context().__eval_context.__contexts[1];
```
Comment (houzefa-abba) [inline web_action_conditionable/static/src/js/views.js:13]: This data getter is needed to get the data. The fact that it is missing explains why the expr eval fails and why `state=='draft'` had to be changed to `'state'=='draft'` (which as @Yajo wrote above, cannot possibly work).
